.editorconfig

```diff
@@ -0,0 +1,20 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+
+[*.{cpp,py,ts}]
+indent_style = space
+indent_size = 4
```

Comment:
4 spaces are consistently used in the code snippets, so this should be uncontroversial.